tests/e2e/test_snowflake_notebooks.py

```diff
@@ -0,0 +1,48 @@
+from unittest import main
+import uuid
+
+from tests.util.snowflake_test_case import SnowflakeTestCase
+
+_STAGE_NAME = "SNOWFLAKE_NOTEBOOKS"
+
+
+class TestSnowflakeNotebooks(SnowflakeTestCase):
+    def test_simple_notebook(self) -> None:
+        self._upload_and_run_notebook(
+            "test_simple_notebook",
+            "simple",
+            "tests/e2e/data/",
+        )
+
+    def _upload_and_run_notebook(
+        self, schema_base_name: str, name: str, path: str
+    ) -> None:
+        try:
+            schema_name = (
+                f"{schema_base_name}_{str(uuid.uuid4()).replace('-', '_')}"
+            )
+            self.create_and_use_schema(schema_name)
+            self.run_query(f"CREATE STAGE {_STAGE_NAME}")
+            self.run_query(
+                f"PUT file://{path}/{name}.ipynb @{_STAGE_NAME} AUTO_COMPRESS = FALSE"
+            )
+            self.run_query(
+                f"PUT file://{path}/environment.yml @{_STAGE_NAME} AUTO_COMPRESS = FALSE"
+            )
+            self.run_query(
+                f"""
+                CREATE NOTEBOOK {name}
+                FROM '@{_STAGE_NAME}'
+                MAIN_FILE = '{name}.ipynb'
+                QUERY_WAREHOUSE = {self._snowflake_connection_parameters["warehouse"]}
+                """
+            )
+            self.run_query(f"ALTER NOTEBOOK {name} ADD LIVE VERSION FROM LAST")
+            self.run_query(f"EXECUTE NOTEBOOK {name}()")
```

Comment:
how does this handle notebook errors?

Reply:
The error is propagated as an `snowflake.snowpark.exceptions.SnowparkSQLException` but the error message will say what was wrong. Here's an example of me running it where I didn't install the right packages:
```
E       snowflake.snowpark.exceptions.SnowparkSQLException: (1304): 01b816c0-0004-1bcf-0038-7187e48d721a: 092848 (42601): User app execution error Python Interpreter Error:
E       Exception: ModuleNotFoundError: Line 1: Module Not Found: trulens.connectors. To import packages from Anaconda, install them first using the package selector at the top of the page.
E       .

../../../../miniconda3/envs/trulens_3_11/lib/python3.11/site-packages/snowflake/connector/errors.py:215: SnowparkSQLException
```